packages/arb-token-bridge-ui/src/components/App/App.tsx

```diff
@@ -238,14 +204,35 @@ const Injector = ({ children }: { children: React.ReactNode }): JSX.Element => {
 
 function NetworkReady({ children }: { children: React.ReactNode }) {
   const [{ l2ChainId }] = useArbQueryParams()
+  const [tosAccepted, setTosAccepted] =
+    useLocalStorage<string>(TOS_LOCALSTORAGE_KEY)
+  const [welcomeDialogProps, openWelcomeDialog] = useDialog()
+  const isTosAccepted = tosAccepted !== undefined
+
+  useEffect(() => {
+    if (!isTosAccepted) {
+      openWelcomeDialog()
+    }
+  }, [isTosAccepted, openWelcomeDialog])
+
+  function onClose(confirmed: boolean) {
+    // Only close after confirming (agreeing to terms)
+    if (confirmed) {
+      setTosAccepted('true')
+      welcomeDialogProps.onClose(confirmed)
+    }
+  }
 
   return (
-    <NetworksAndSignersProvider
-      selectedL2ChainId={l2ChainId || undefined}
-      fallback={fallbackProps => <ConnectionFallback {...fallbackProps} />}
-    >
-      {children}
-    </NetworksAndSignersProvider>
+    <>
+      <WelcomeDialog {...welcomeDialogProps} onClose={onClose} />
```

Comment:
moved `WelcomeDialog` here to remove conditional in `App`
also to show a more consistent background because on production we don't show the header or background before if user is connected but didn't accept ToS